governance/reviews/cloudevents-08-29-2023.md

```diff
@@ -0,0 +1,210 @@
+
+Transparency: The project has attempted to add every process and procedure to written governance files.  Meetings and votes are well-documented, aside from the SDKs.  As a spec project, CloudEvents is clearly trying to be fully transparent.  However, governance documentation is confusingly organized to the point of impenetrability; there is no way that a casual observer or contributor would be able to participate in, or even monitor, the project's activities and decisions without direct mentoring. For example, lists of Admins and Voting members include minimal information and can be hard to decipher. This problem could be solved with a refactor of the governance documentation and adding some things like a high-level explanation of the project's organization and a public calendar.  This opaqueness appears to be purely a documentation problem and not an execution problem.
+
+Freshness: CloudEvents has updated their governance docs multiple times in the project history.  The last batch of governance updates was 8 months ago.
+
+#### Governance Drift
+
+<!---  Are the governance activities being carried out? Are community meetings (if any) happening? Are required elections and votes taking place? Are official communications channels accessible, staffed and responsive? Are they being used? Are questions and proposed updates/changes to governance (if any) being transparently discussed and addressed? -->
+
+Governance Drift can occur when the executed and observable governance of a project deviates from the documented governance of the project.
+
+Governance drift for CloudEvents cannot be determined because the written governance is too hard to decipher.
+
+#### Ownership
+
+The main project uses OWNERS files which appear to match declare roles elsewhere.  At this time, we cannot audit actual Github permissions.
+
+The SDKs do not appear to have any controls on permissions, and as such are likely very out of sync with project roles.
+
+### Maintainer List(s)
+
+CloudEvents' three types of leaders need to be evaluated separately.
+
+#### Admins
+
+The list of Admins is current, and appears well-balanced, with four admins representing four different organizations.  However, admins are listed only by GH handle, with no affiliation, contact information, or history; listing full information on each admin would enhance transparency.  The Admins were set up as roles-for-life, with only one change of admin since the project was founded in 2018.  There are no defined processes for removing an admin for inactivity, nor any expectation that new Admins will be recruited.
+
+Since, at this time one of the four admins (Ken Owens) is no longer active in the project per Devstats and meeting tracking, it would benefit Cloud Events to document a full Admin lifecycle including recruitment, promotion, qualifications, and emeritus status.  Documenting this and following it would help ensure that the project retains a continuously active leadership.
```

Comment:
Perhaps we need some more text around this, but it's a bit misleading to call it "leadership" since they really only act after approval from the group. They're more like secretaries.

Reply:
That isn't clear from the documentation; on a reading from the docs the admins are treated as Maintainers.  Who are the Maintainers, but the CNCF definition?  Can we make that clear?

Reply:
https://github.com/cloudevents/spec/blob/main/docs/GOVERNANCE.md#admins
says:
> Since the role of an 'Admin' is mainly administrative...

And there's also this text under the definition of admin:
> Admins are Members of the group but have the ability to perform administrative actions on behalf of the group. For example, manage the website, github repos and moderate the meetings. Their actions should be done with the knowledge and consent of the group. They also have the ability to merge/close PRs, but only per the group's approval.

Reply:
So it's nice to think of the admins as servants of the voting members, but the admins actually have quite a bit of power over the project.  As such, they need to have a clearly defined lifecycle, just like we require regular code projects to have a clearly defined lifecycle for their maintainers.

Reply:
I'm still confused. We define how admins can be added and removed - that's the lifecycle: https://github.com/cloudevents/spec/blob/main/docs/GOVERNANCE.md#admins It's not clear to me what's missing.